client/my-sites/email/email-providers-in-depth-comparison/types.ts

```diff
@@ -18,6 +22,7 @@ export type EmailProviderFeatures = {
 export type EmailProvidersInDepthComparisonProps = {
 	comparisonContext: string;
 	selectedDomainName: string;
+	selectedIntervalLength: string;
```

Comment:
Why not require this to be an `IntervalLength` to start with? That means we can push the `castIntervalLength()` responsibility out to the caller rather than having the component enforce types later on.